### ISSUE TITLE
chat.octet.model.exceptions.ModelException: Loading native library error 

### ISSUE THREAD

**Issue by Lelalue** [eoctet/Octet.Chat#2]
Getting error when initating the model 
like : 

` ModelParameter modelParams = ModelParameter.builder()
                .modelPath(modelPath)
                .threads(6)
                .contextSize(4096)
                .verbose(true)
                .gpuLayers(10)
                .build();`

        model = new Model(modelParams);`
        
 Exception is thrown in line 80 of Model.java
 
 `at org.springframework.context.event.ApplicationListenerMethodAdapter.doInvoke(ApplicationListenerMethodAdapter.java:365) ~[spring-context-6.0.13.jar:6.0.13]
	at org.springframework.context.event.ApplicationListenerMethodAdapter.processEvent(ApplicationListenerMethodAdapter.java:233) ~[spring-context-6.0.13.jar:6.0.13]
	at org.springframework.context.event.ApplicationListenerMethodAdapter.onApplicationEvent(ApplicationListenerMethodAdapter.java:165) ~[spring-context-6.0.13.jar:6.0.13]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:174) ~[spring-context-6.0.13.jar:6.0.13]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:167) ~[spring-context-6.0.13.jar:6.0.13]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:145) ~[spring-context-6.0.13.jar:6.0.13]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:445) ~[spring-context-6.0.13.jar:6.0.13]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:378) ~[spring-context-6.0.13.jar:6.0.13]
	at org.springframework.boot.context.event.EventPublishingRunListener.started(EventPublishingRunListener.java:103) ~[spring-boot-3.1.5.jar:3.1.5]
	at org.springframework.boot.SpringApplicationRunListeners.lambda$started$5(SpringApplicationRunListeners.java:76) ~[spring-boot-3.1.5.jar:3.1.5]
	at java.base/java.lang.Iterable.forEach(Iterable.java:75) ~[na:na]
	at org.springframework.boot.SpringApplicationRunListeners.doWithListeners(SpringApplicationRunListeners.java:118) ~[spring-boot-3.1.5.jar:3.1.5]
	at org.springframework.boot.SpringApplicationRunListeners.doWithListeners(SpringApplicationRunListeners.java:112) ~[spring-boot-3.1.5.jar:3.1.5]
	at org.springframework.boot.SpringApplicationRunListeners.started(SpringApplicationRunListeners.java:76) ~[spring-boot-3.1.5.jar:3.1.5]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:322) ~[spring-boot-3.1.5.jar:3.1.5]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1306) ~[spring-boot-3.1.5.jar:3.1.5]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1295) ~[spring-boot-3.1.5.jar:3.1.5]
	at de.bitsea.runLlama.RunLlama.main(RunLlama.java:31) ~[main/:na]
Caused by: java.lang.ExceptionInInitializerError: null
	at chat.octet.model.LlamaService.<clinit>(LlamaService.java:28) ~[llama-java-core-1.2.7.jar:na]
	at chat.octet.model.Model.getLlamaModelParameters(Model.java:80) ~[llama-java-core-1.2.7.jar:na]
	at chat.octet.model.Model.<init>(Model.java:57) ~[llama-java-core-1.2.7.jar:na]
	at de.bitsea.runLlama.LlamaModelService.init(LlamaModelService.java:34) ~[main/:na]
	at de.bitsea.runLlama.RunLlama.printApplicationUrl(RunLlama.java:38) ~[main/:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:568) ~[na:na]
	at org.springframework.context.event.ApplicationListenerMethodAdapter.doInvoke(ApplicationListenerMethodAdapter.java:348) ~[spring-context-6.0.13.jar:6.0.13]
	... 17 common frames omitted
Caused by: chat.octet.model.exceptions.ModelException: Loading native library error 
	at chat.octet.model.utils.Platform.loadLibraryResource(Platform.java:123) ~[llama-java-core-1.2.7.jar:na]
	at chat.octet.model.utils.Platform.<clinit>(Platform.java:221) ~[llama-java-core-1.2.7.jar:na]
	... 27 common frames omitted
Caused by: java.lang.NullPointerException: null
	at java.base/java.util.Objects.requireNonNull(Objects.java:208) ~[na:na]
	at chat.octet.model.utils.Platform.loadLibraryResource(Platform.java:114) ~[llama-java-core-1.2.7.jar:na]`

**Comment by eoctet** [comment]
@Lelalue 

> The reason for NPE is due to an error in loading libllama file. 
```java
       // chat.octet.model.utils.Platform.loadLibraryResource: 114
        try (InputStream in = Platform.class.getResourceAsStream(File.separator + libraryPath + File.separator + libraryName);
             BufferedInputStream reader = new BufferedInputStream(Objects.requireNonNull(in));
             FileOutputStream out = new FileOutputStream(libraryFile)
        ) {
            byte[] buf = new byte[1024];
            while (reader.read(buf) > 0) {
                out.write(buf);
                Arrays.fill(buf, (byte) 0);
            }
        } catch (Exception e) {
            throw new ModelException("Loading native library error ", e);
        }
```

What version of your OS is it?

**Comment by ghost** [comment]
@eoctet 
im getting a similar Error while trying to instantiate the Model -> this.llamaModel = new Model(modelParams);

stack-trace:

...
Caused by: java.lang.ExceptionInInitializerError
	at chat.octet.model.LlamaService.<clinit>(LlamaService.java:28)
	at chat.octet.model.Model.getLlamaModelParameters(Model.java:80)
	at chat.octet.model.Model.<init>(Model.java:57)
	at com.systems.ow.service.impl.LLaMaAnswerRetrievalServiceImpl.init(LLaMaAnswerRetrievalServiceImpl.java:29)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleMethod.invoke(InitDestroyAnnotationBeanPostProcessor.java:457)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleMetadata.invokeInitMethods(InitDestroyAnnotationBeanPostProcessor.java:401)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeInitialization(InitDestroyAnnotationBeanPostProcessor.java:219)
	... 109 more
Caused by: chat.octet.model.exceptions.ModelException: Loading native library error 
	at chat.octet.model.utils.Platform.loadLibraryResource(Platform.java:123)
	at chat.octet.model.utils.Platform.<clinit>(Platform.java:221)
	... 118 more
Caused by: java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:233)
	at chat.octet.model.utils.Platform.loadLibraryResource(Platform.java:114)
	... 119 more

Im operating on a windows 11 enterprise 64


private static final String MODEL_FILE = "llama2/llama-2-7b.Q5_K_M.gguf";
    private Model llamaModel;

    @PostConstruct
    public void init() {
        try {
            String modelPath = new ClassPathResource(MODEL_FILE).getFile().getAbsolutePath();
            ModelParameter modelParams = ModelParameter.builder()
                    .modelPath(modelPath)
                    .threads(6)
                    .contextSize(4096)
                    .verbose(true)
                    .build();

            this.llamaModel = new Model(modelParams);
        } catch (IOException e) {
            throw new RuntimeException("Failed to initialize LLaMa model", e);
        }
    }

**Comment by eoctet** [comment]
@Lelalue @Phalanx-SoftDev 

Hi, guys!
I have fixed this issue in v1.2.9 😃
Please update to latest version ~


```xml
<dependency>
    <groupId>chat.octet</groupId>
    <artifactId>llama-java-core</artifactId>
    <version>1.2.9</version>
</dependency>
```
